include/sympack/symPACKMatrix2D.hpp

```diff
@@ -1005,26 +1228,105 @@ namespace symPACK{
             }
             else {
               //Compute the update in a temporary buffer
-//#ifdef SP_THREADS
               tmpBuffers.tmpBuf.resize(tgt_width*src_nrows + src_snode_size*tgt_width);
-//#endif
               buf = &tmpBuffers.tmpBuf[0];
             }
 
             if ( in_place && this->i == this->j ) {
               bassert(src_nrows==tgt_width);
               SYMPACK_TIMER_SPECIAL_START(UPDATE_SNODE_SYRK);
+#ifdef CUDA_MODE
+              if (tgt_width * src_snode_size > symPACK::syrk_limit) {
+            
+                dev_ptr d_pivot_nzval = symPACK::gpu_allocator.allocate<T>(tgt_width * src_snode_size);
+                dev_ptr d_buf = symPACK::gpu_allocator.allocate<T>(tgt_width * ldbuf);
+                
+                if (d_pivot_nzval==nullptr) {
+                    blas::Syrk('U','T',tgt_width, src_snode_size, T(-1.0), pivot_nzval, src_snode_size, beta, buf, ldbuf);
+                } else if (d_buf==nullptr) { 
+                    blas::Syrk('U','T',tgt_width, src_snode_size, T(-1.0), pivot_nzval, src_snode_size, beta, buf, ldbuf);
```

Comment:
Similar observation here - the first conditional path above can sometimes leak `d_buf`, and the second is guaranteed to leak `d_pivot_nzval`